### PR TITLE
[16.0][IMP] resource_booking: Add search to partner_id field to prevent "wrong" results

### PR DESCRIPTION
**Pull request by victoralmau**
Add search to partner_id field to prevent "_wrong_" results

Please @pedrobaeza can you review it?

@Tecnativa TT50035

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Add also the noupdate changes to the migration script for completeness.

File: resource_booking/models/resource_booking.py [hunk 1]
```diff
@@ -270,6 +271,14 @@ def _inverse_partner_id(self):
         for one in self:
             one.partner_ids = one.partner_id
 
+    @api.model
+    def _search_partner_id(self, operator, value):
+        """Overwrite this for security, partner_id field is not store and if we search
```
Comment (pedrobaeza) [inline resource_booking/models/resource_booking.py]:
```suggestion
        """Overwrite this for security, partner_id field is not stored and if we search
```

**Comment by victoralmau** [comment]
> Add also the noupdate changes to the migration script for completeness.

I will add it in v15 (when this is merged, I will backport to v15).

**Review by pedrobaeza — APPROVED** [review]

**Review by carolinafernandez-tecnativa — APPROVED** [review]
Functional review, LGTM :)
Thanks!

**Pull request merged**